filters/context_test.go

```diff
@@ -0,0 +1,47 @@
+package filters
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestContext(t *testing.T) {
+	ctx := context.WithValue(context.WithValue(context.Background(), 1, "one"), "b", "bee")
+	adapted := AdaptContext(ctx).WithValue("c", "cee")
+	require.Equal(t, "one", adapted.Value(1))
+	require.Equal(t, "bee", adapted.Value("b"))
+	require.Equal(t, "cee", adapted.Value("c"))
+	adapted = adapted.WithValue("c", "cee2")
+	require.Equal(t, "cee2", adapted.Value("c"))
+	require.Equal(t, nil, adapted.Value("d"))
+}
+
+func BenchmarkRegularContext(b *testing.B) {
+	rootKey := "root"
+	key := "key"
+	ctx := context.WithValue(context.Background(), rootKey, "r")
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		ctx = context.WithValue(ctx, key, i)
+		for j := 0; j < 10; j++ {
+			ctx.Value(key)
+			ctx.Value(rootKey)
+		}
+	}
```

Comment:
This is kind of nit-picky, but these benchmarks aren't going to give you an accurate idea of the performance differences.  The time taken per op will scale with `b.N`, which shouldn't really be the case.  I think you'd want something like:
```
ctx := context.Background()
for i := 0; i < 1000; i++ {
  ctx = context.WithValue("some key", "some value")
}
b.ResetTimer()
for i := 0; i < b.N; i++ {
  ctx.Value("some key")
}
```

That would specifically test the difference across `WithValue` for a context with 1000 insertions.

But... I don't think it's necessary to have benchmarks at all.  Your implementation is clearly amortized O(1) vs the standard library's O(n).